doc/src/modules/physics/continuum_mechanics/beam_problems.rst

```diff
@@ -44,6 +44,17 @@ Examples
 ========
 
 Below are examples of a variety two dimensional beam bending problems.
+ 
+.. note::
+
+    There is no specific sign convention that must be followed. The only thing
+    that should be considered while choosing sign convention is that
+    upward forces and clockwise moments have the same signs.
```

Comment:
Can you formulate this statement in terms of a right-handed coordinate system? "upward forces and clockwise moments" is not quite precise enough language. For example something more like "Given a right handed coordinate system with XYZ coordinates, the beam's length is assumed to be along the X axis. The loading forces are directed in the positive or negative Y direction and the loading moments are about the positive or negative Z direction. The sign convention has to ..."

Reply:
Secondly, why is that there a convention restriction relating the conventions of forces and moments? Shouldn't the user be able to choose what they want. What in the beam code is forcing this convention?

Reply:
> Secondly, why is that there a convention restriction relating the conventions of forces and moments? Shouldn't the user be able to choose what they want. What in the beam code is forcing this convention?

The beam code calculates moment at the rightmost point of the beam. This causes upward forces to generate clockwise moment. Hence they should be of the same sign.

Reply:
I'm not sure that's how sign conventions work. The user should be able to specify in their problem set what is a positive direction for forces and what a positive direction for moments are. There is no reason that upward forces should cause a specific sign conventions in the moments.

Now, it is possible that the underlying code does make some assumed sign convention. If it does, what precisely is that assumption? And where in the code is that assumption enforced?

If the above is true, we should do one of two things to solve it: 1) remove the assumption and let the user set whatever assumptions about sign convention they desire or 2) articulate the assumption in the documentation.

Before I approve all these changes to the documentation, we need to know that this is the correct change and why. It is not clear, to me, what precisely is being fixed here.

Reply:
> Can you formulate this statement in terms of a right-handed coordinate system?

I will update it.


> I'm not sure that's how sign conventions work. The user should be able to specify in their problem set what is a positive direction for forces and what a positive direction for moments are. There is no reason that upward forces should cause a specific sign conventions in the moments.

The code calculates moment at the rightmost point of the beam. It does so by multiplying all the forces with the distance from the right most point. We know that 
Moment _**M = r**_ x _**F**_
Here _**r**_ vector points in **_-x_** direction and an upward force points in **_+y_** direction. The resulting moment will therefore have **_-z_** direction which is same as that of a clockwise moment.
Hence when moment equation is formed all the moments pointing in a particular direction should have same signs.
These equations are generated here in the solve_for_reaction_loads function in beam.py
`shear_curve = limit(self.shear_force(), x, l)
`
`
moment_curve = limit(self.bending_moment(), x, l)
` 
> 
> Now, it is possible that the underlying code does make some assumed sign convention. If it does, what precisely is that assumption? And where in the code is that assumption enforced?

The code does not make any assumptions in sign convention, it just calculates moment at the rightmost point.
A force can cause moments in different directions depending on the point about which we are calculating moment. The externally applied moment however does not depend upon the point in consideration.
[https://pages.mtu.edu/~dnaneet/2150/notes/problems/sfd_bmd_cantilever_UDL_Moment_PtLoad](url) 
See this. In this example, if we calculate moment about A, the 40 kN force will cause a moment in clockwise direction but 50 kN-m will act in anticlockwise direction as given. Hence they have opposite signs. However if we calculate moment about point B (which is what the code does), the moment caused by 40 kN force will be in anticlock wise direction and so will the 50 kN-m moment.
The moment eq about B would look like
40*(6.5) + 50 + Ma - 52*9 = 0 (This is what the code does)
which gives Ma = +158
The positive sign here represents anticlockwise moment. Notice how downward force is taken positive in this case.
The sign convention of moments depend upon the direction of forces only because we take moment at the rightmost point.

> If the above is true, we should do one of two things to solve it: 1) remove the assumption and let the user set whatever assumptions about sign convention they desire or 2) articulate the assumption in the documentation.

I don't think its a restriction or something that needs to be fixed. I think a note regarding this should be enough so that the user knows we are taking moment at the rightmost point and enter inputs accordingly.

Reply:
> The code calculates moment at the rightmost point of the beam.

What code does? The user constructs singularity functions which are not tied to whether the origin is at the "right" or "left". In fact, "right" and "left" have no meaning in this context.

I don't see how the point at which one evaluates a moment due to a force has any bearing on the sign convention. The sign convention of the moment only tells one if the moment is positive about the positive or negative Z axis. It may be true that the code incorrectly calculates moments, but that should be independent of sign conventions, i.e. the absolute value of a moment at any point should be independent of sign conventions.

Where in the underlying `Beam()` code is any assumption made about sign conventions? What method makes this assumption? Is it the method that solves for the reactions?

Reply:
> What code does? The user constructs singularity functions which are not tied to whether the origin is at the "right" or "left". In fact, "right" and "left" have no meaning in this context.

The user constructs singularity functions which are with respect to the origin and distance x from it.
The code written in `solve_for_reaction_loads` applies limit x tending to L to the singularity function of `bending_moment` in beam.py
`moment_curve = limit(self.bending_moment(), x, l)` This statements does that.
When this limit is applied the resulting equation becomes the moment equation at point x = L, which is the rightmost point of the beam.
 
> I don't see how the point at which one evaluates a moment due to a force has any bearing on the sign convention. The sign convention of the moment only tells one if the moment is positive about the positive or negative Z axis. It may be true that the code incorrectly calculates moments, but that should be independent of sign conventions, i.e. the absolute value of a moment at any point should be independent of sign conventions.

Yes the absolute value of moment at any point should be independent of sign conventions.
But moments acting in same direction( +z or -z) should have same signs, otherwise there will be errors in calculations. We have to make sure that the moments caused by forces along a certain direction and moments acting externally along that direction have the same signs.
This is can only be done when we input forces causing moment along a certain direction and other moments acting along that direction with same signs.
  
> Where in the underlying `Beam()` code is any assumption made about sign conventions? What method makes this assumption? Is it the method that solves for the reactions?

The code does not make any sign conventions. The solve_for_reaction_loads function simply applies limit to the bending moment eq thereby calculating moment at x=L

Reply:
>  This is can only be done when we input forces causing moment along a certain direction and other moments acting along that direction with same signs.

So is this only a user input error? Is the user somehow using two different sign conventions for forces or two different ones for moment?

Reply:
> So is this only a user input error? Is the user somehow using two different sign conventions for forces or two different ones for moment?

Yes, apart from adding the note i also changed the sign of input load in example 1 in earlier commits. The user was using different sign conventions for moments acting along +z direction and the moments caused by forces that act in +z direction as well.
This user error can be avoided by adding the note.
The code does not assume any sign convention but we have to be careful while inputting loads and moments so that moments acting along a certain direction end up having same signs(they can be either positve or negative).

Reply:
Please review @moorepants